### PR TITLE
Spaces link modals

### PR DESCRIPTION
**Pull request by stevenviar**
I'll add more tests for the modals, but wanted to get this out there and see if anyone has any concerns/feedback. This is based on an older alpha version of Spaces currently used by one of our apps which needs the modal functionality. 

**Before submitting a pull request,** please make sure the following is done:

1. Fork [the repository](https://github.com/availity/availity-react) and create your branch from `master`.
2. Run `yarn` in the repository root.
3. If you've fixed a bug or added code that should be tested, add tests!
4. Ensure the test suite passes (`yarn test`). Tip: `yarn test --watch TestName` is helpful in development.
5. Make sure your code passed the conventional commits check. Read more about [conventional commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)

### CODE REVIEW

File: packages/spaces/src/modals/DisclaimerModal.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    ... on Text {
+      description
+    }
+  }
+}`;
+
+const DisclaimerModal = ({ disclaimerId }) => {
+  const [disclaimer, setDisclaimer] = useState('');
+
+  const fetchDisclaimer = async () => {
```
Comment (jordan-a-young) [inline packages/spaces/src/modals/DisclaimerModal.js]: I think we should change the `useEffectAsync` to `useEffect` and move this function inside of there

**Review by jordan-a-young — CHANGES_REQUESTED** [review]

File: packages/spaces/src/modals/ModalProvider.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    <ModalContext.Provider
+      value={(modalType, modalOptions) =>
+        dispatch({
+          type: `OPEN_${modalType}`,
+          ...modalOptions,
+        })
+      }
+    >
+      <Modal toggle={toggle} isOpen={isOpen}>
+        <ModalHeader>{title}</ModalHeader>
```
Comment (LauRoxx) [inline packages/spaces/src/modals/ModalProvider.js]: The reactstrap modal needs a little more to be compliant. Example code snippet I pulled from the Dialog Structure story on the ADA Board:
`<Modal
   labelledBy="feedback-form-header"
   role="dialog"
   aria-modal>`
    `<ModalHeader
        id="feedback-form-header">`

Comment (LauRoxx) [reply packages/spaces/src/modals/ModalProvider.js]: forgot that reactstrap modalheader is an h5 which is not correct for modals. reactstrap lets you change the modal-title tag but classnames are passed to the outer div. This is the workaround that was done for feedback to keep the h5 styling.

`<ModalHeader id="feedback-form-header"
        role="heading"
        aria-level="2"
        className="h5"
        tag="div">`

or you could probably also do 
`<ModalHeader tag="div"><h2 className="h5">{title}</h2></ModalHeader>`

**If your modal has the "X" to close it then stick with the second option.

Comment (stevenviar) [reply packages/spaces/src/modals/ModalProvider.js]: Thanks for the examples, updated and think everything should be resolved now.

**Review by jordan-a-young — APPROVED** [review]

**Review by gregmartDOTin — APPROVED** [review]

**Pull request merged**